dvc/command/ls.py

```diff
@@ -1,28 +1,45 @@
 import argparse
 import logging
+from functools import partial
 
 from dvc.command.base import append_doc_link
 from dvc.command.base import CmdBaseNoRepo
 from dvc.exceptions import DvcException
+from dvc.logger import LoggerHandler
 
 
 logger = logging.getLogger(__name__)
 
 
 class CmdList(CmdBaseNoRepo):
+    def get_log_fn(self, log_level=logging.INFO):
+        dvc_logger = logging.getLogger("dvc")
+        with_color = False
+
+        handlers = sorted(dvc_logger.handlers, key=lambda h: h.level)
+        handler = next((h for h in handlers if h.level <= log_level), False)
+        if handler:
+            with_color = (
+                isinstance(handler, LoggerHandler) and handler.stream.isatty()
+            )
+
+        return (with_color, partial(logger.log, log_level))
+
     def run(self):
         from dvc.repo import Repo
 
+        (with_color, log) = self.get_log_fn()
         try:
-            nodes = Repo.ls(
+            fs_nodes = Repo.ls(
                 self.args.url,
                 self.args.target,
                 rev=self.args.rev,
                 recursive=self.args.recursive,
                 outs_only=self.args.outs_only,
+                with_color=with_color,
             )
-            if nodes:
-                logger.info("\n".join(nodes))
+            if fs_nodes:
+                log("\n".join(fs_nodes))
```

Comment:
Not sure I understand why we choose `log` instead of continuing dumping into `logger.info` as we do in all other places?

Reply:
I need to check - the related logger's handler is `tty`, so handlers are depends on logger and log level, so in `self.get_log_fn` I got that info as
```
(with_color, log) = self.get_log_fn(logging.INFO)
```
 I can make it as
```
def hasatty_handler(logger, loglevel):
   ...

with_color = hasatty_handler(logger, logging.INFO)
...
logger.info(...)
```
but it splits two dependent sections and introduce place for an error

Reply:
Now I've got possible problem if we have different handlers for the same log level: for example file and stdout. In that case the output will contain color codes
What do you think, @efiop, maybe better to support `color` with additional option (`-C/--color`)?

Reply:
@JIoJIaJIu Ah, got it. Good point! If you look into other occurrences of `isatty` in our code, you'll see that we usually do some pretty bald assumptions and would generally just check sys.stdout.isatty() here and would just run with it. Even though, strictly speaking, that is not 100% correct, but for the sake of consistency let's do the same here. We have had lots of discussions about stopping abusing `logger` for things that `print` should handle, but it is better to do those in one sitting later.